app/forms/mappings/Constraints.scala

```diff
@@ -20,6 +20,8 @@ import play.api.data.validation.{Constraint, Invalid, Valid}
 
 trait Constraints {
 
+  val crn = """^\d{7}|[a-zA-Z]{1,2}\d{6}$"""
```

Comment:
This should either be private if possible or package private it we want it for tests.

Reply:
Protected works. It's referenced in classes that extend the trait.